File: PRACTICA2/QytetetJava/src/modeloqytetet/Jugador.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package modeloqytetet;

import java.util.ArrayList;

/**
 *
 * @author jesus
 */
public class Jugador {
    private boolean encarcelado;
    private String nombre;
    private int saldo;
    private Sorpresa cartaLibertad;
    private Casilla casillaActual;
    private ArrayList<TituloPropiedad> propiedades;

    public Jugador(String nombre) {
        this.nombre = nombre;
        this.encarcelado = false;
        this.saldo = 7500;
        
    }
    
    
    
    boolean cancelarHipoteca(TituloPropiedad titulo)
    {
        throw new UnsupportedOperationException("Sin implementar");
    }
    
    boolean comprarTituloPropiedad()
    {
        throw new UnsupportedOperationException("Sin implementar");
    }
    
    int cuantasCasasHotelesTengo()
    {
        throw new UnsupportedOperationException("Sin implementar");
    }
    
    boolean deboPagarAlquiler()
    {
        throw new UnsupportedOperationException("Sin implementar");
    }
    
    Sorpresa devolverCartaLibertad()
    {
        throw new UnsupportedOperationException("Sin implementar");
    }
    
    boolean edificarCasa(TituloPropiedad titulo)
    {
        throw new UnsupportedOperationException("Sin implementar");
    }
    
    boolean edificarHotel(TituloPropiedad titulo)
    {
        throw new UnsupportedOperationException("Sin implementar");
    }
    
    private void eliminarDeMisPropiedades(TituloPropiedad titulo)
    {
        throw new UnsupportedOperationException("Sin implementar");
    }
    
    private boolean esDeMiPropiedad(TituloPropiedad titulo)
    {
        throw new UnsupportedOperationException("Sin implementar");
    }
    
    boolean estoyEnCalleLibre()
    {
        throw new UnsupportedOperationException("Sin implementar");
    }
    
    boolean hipotecarPropiedad(TituloPropiedad titulo)
    {
        throw new UnsupportedOperationException("Sin implementar");
    }
    
    void irACarcel(Casilla casilla)
    {
        throw new UnsupportedOperationException("Sin implementar");
    }
    
    int modificarSaldo(int cantidad)
    {
        throw new UnsupportedOperationException("Sin implementar");
    }
    
    int obtenerCapital()
    {
        throw new UnsupportedOperationException("Sin implementar");
    }
    
    ArrayList<TituloPropiedad> obtenerPropiedades(boolean hipotecada)
    {
        throw new UnsupportedOperationException("Sin implementar");
    }
    
    void pagarAlquiler()
    {
        
    }
    
    void pagarImpuesto()
    {
        
    }
    
    void pagarLibertad(int cantidad)
    {
        
    }
    
    boolean tengoCartaLibertad()
    {
        throw new UnsupportedOperationException("Sin implementar");
    }
    
    private boolean tengoSaldo(int cantidad)
    {
        throw new UnsupportedOperationException("Sin implementar");
    }
    
    boolean venderPropiedad(Casilla casilla)
    {
        throw new UnsupportedOperationException("Sin implementar");
    }
    
    
    //CONSULTODRES Y MODIFICADORES

    boolean isEncarcelado() {
        return encarcelado;
    }

    String getNombre() {
        return nombre;
    }

    public int getSaldo() {
        return saldo;
    }

    Sorpresa getCartaLibertad() {
        return cartaLibertad;
    }

    Casilla getCasillaActual() {
        return casillaActual;
    }

    ArrayList<TituloPropiedad> getPropiedades() {
        return propiedades;
    }

    void setEncarcelado(boolean encarcelado) {
        this.encarcelado = encarcelado;
    }

    void setCartaLibertad(Sorpresa cartaLibertad) {
        this.cartaLibertad = cartaLibertad;
    }

    void setCasillaActual(Casilla casillaActual) {
        this.casillaActual = casillaActual;
    }

    @Override
    public String toString() {
        return "Jugador{" + "encarcelado=" + encarcelado + ", nombre=" + nombre + ", saldo=" + saldo + ", cartaLibertad=" + cartaLibertad + ", casillaActual=" + casillaActual + ", propiedades=" + propiedades + '}';
    }

    
    
    
    
    
    
}

File: PRACTICA2/QytetetRuby/lib/jugador.rb
#encoding: utf-8
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.

module ModeloQytetet
  class Jugador
    
    attr_accessor :nombre, :encarcelado, :saldo, :cartaLibertad, :casillaActual, :propiedades
    
    def initialize(nombre)
      @nombre = nombre;
      @encarcelado = false
      @saldo = 7500
      @cartaLibertad = nil
      @casillaActual = nil
      @propiedades = nil
    end
    
    def cancelarHipoteca(titulo)
    
        raise NotImplementedError
    end
    
    def comprarTituloPropiedad
    
        raise NotImplementedError
    end
    
    def cuantasCasasHotelesTengo
    
        raise NotImplementedError
    end
    
    def deboPagarAlquiler
    
        raise NotImplementedError
    end
    
    def devolverCartaLibertad
    
        raise NotImplementedError
    end
    
    def edificarCasa(titulo)
    
        raise NotImplementedError
    end
    
    def edificarHotel(titulo)
    
        raise NotImplementedError
    end
    
    def eliminarDeMisPropiedades(titulo)
    
        raise NotImplementedError
    end
    
    def esDeMiPropiedad(titulo)
    
        raise NotImplementedError
    end
    
    def estoyEnCalleLibre
    
        raise NotImplementedError
    end
    
    def hipotecarPropiedad(titulo)
    
        raise NotImplementedError
    end
    
    def irACarcel(casilla)
    
        raise NotImplementedError
    end
    
    def modificarSaldo(cantidad)
    
        raise NotImplementedError
    end
    
    def obtenerCapital
    
        raise NotImplementedError
    end
    
    def obtenerPropiedades(hipotecada)
    
        raise NotImplementedError
    end
    
    def pagarAlquiler
    
        
    end
    
    def pagarImpuesto
    
        
    end
    
    def pagarLibertad(cantidad)
    
        
    end
    
    def tengoCartaLibertad()
    
        raise NotImplementedError
    end
    
    def tengoSaldo(cantidad)
    
        raise NotImplementedError
    end
    
    def venderPropiedad(casilla)
    
        raise NotImplementedError
    end
    
  end
end
